File: python/hamming/hamming.py
#!/usr/bin/env python3

def distance(strand_a: str, strand_b: str) -> int:
    """Take two strands as strings, and return the hamming distance."""
    if len(strand_a) != len(strand_b):
        raise ValueError("Input must be valid strands of equal length.")
    return len(["Not Equal" for a, b in zip(strand_a, strand_b) if a != b])

File: go/hamming/hamming.go
package hamming

import (
    "errors"
)

func Distance(a, b string) (int, error) {
    if len(a) != len(b) {
        return 0, errors.New("Input strings must be of equal length.")
    }
    ham := 0
    for i := 0; i < len(a); i++ {
        if a[i] != b[i] { ham ++ }
    }
    return ham, nil
}
